infra/managed-cluster/aks-cluster.bicep

```diff
@@ -0,0 +1,187 @@
+@description('Required. The parameter object for the virtual network. The object must contain the name,resourceGroup and subnetClusterNodes values.')
+param vnet object
+
+@description('Required. The parameter object for the cluster. The object must contain the name,skuTier,nodeResourceGroup,miControlPlane,adminAadGroupObjectId and monitoringWorkspace values.')
+param cluster object
+
+@allowed([
+  'UKSouth'
+])
+@description('Required. The Azure region where the resources will be deployed.')
+param location string
+@description('Required. Environment name.')
+param environment string
+@description('Required. Date in the format yyyy-MM-dd.')
+param createdDate string = utcNow('yyyy-MM-dd')
+@description('Required. Date in the format yyyyMMdd-HHmmss.')
+param deploymentDate string = utcNow('yyyyMMdd-HHmmss')
+
+var kubernetesVersion = '1.26.6'
+
+var customTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+}
+var tags = union(loadJsonContent('../default-tags.json'), customTags)
+
+var tagsMi = {
+  Name: cluster.miControlPlane
+  Purpose: 'Managed Identity'
+  Tier: 'Security'
+}
+
+var aksTags = {
+  Name: cluster.name
+  Purpose: 'AKS Cluster'
+  Tier: 'Shared'
+}
+
+resource vnetResourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' existing = {
+  scope: subscription()
+  name: vnet.resourceGroup
+}
+
+resource clusterVirtualNetwork 'Microsoft.Network/virtualNetworks@2023-02-01' existing = {
+  scope: vnetResourceGroup
```

Comment:
Can we not use the resourceid or resourcegroup function to get the resource name here?

Reply:
Changed to resourceGroup function